### PR TITLE
XML.py: Add language definition for XML

### PR DESCRIPTION
**Pull request by virresh**
Closes https://github.com/coala/coala/issues/5692

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: coalib/bearlib/languages/definitions/XML.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from coalib.bearlib.languages.Language import Language
+
+
+@Language
+class XML:
+    __qualname__ = 'Extensible Markup Language'
+    aliases = 'xml',
+
```
Comment (pareksha) [inline coalib/bearlib/languages/definitions/XML.py]: Why blank line here?

File: coalib/bearlib/languages/definitions/XML.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    extensions = '.xml',
+    multiline_comment_delimiters = {'<!--': '-->'}
+    string_delimiters = {'"': '"', "'": "'"}
+    multiline_string_delimiters = string_delimiters
+    encapsulators = {'<': '>'}
```
[the remark this answers is not in the document]
Comment (virresh) [reply coalib/bearlib/languages/definitions/XML.py]: @jayvdb @pareksha 
I referred https://github.com/coala/coala/blob/master/coalib/bearlib/languages/definitions/html.py
for referring to how we use markups in language definitions,
I am not particularly sure of how these are used by core 
Could only find one place in bears repo (the IndentationBear) where these were used, and considering that, I believe these are also sufficient to determine indents of tags ?


Also @pareksha , `</>` is a subset of `<>`, note the same literal opens and ends tag

File: coalib/bearlib/languages/definitions/XML.py (hunk 1 above)
Comment (virresh) [reply coalib/bearlib/languages/definitions/XML.py]: c.f https://github.com/coala/coala/blob/master/coalib/bearlib/languages/definitions/html.py#L8

File: coalib/bearlib/languages/definitions/XML.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+class XML:
+    __qualname__ = 'Extensible Markup Language'
+    aliases = 'xml',
+    versions = 1.0,
+
+    extensions = '.xml',
+    multiline_comment_delimiters = {'<!--': '-->'}
+    string_delimiters = {'"': '"', "'": "'", '<![CDATA[': ']]>'}
+    multiline_string_delimiters = string_delimiters
+    encapsulators = {'<': '>'}
```
Comment (jayvdb) [inline coalib/bearlib/languages/definitions/XML.py]: I suspect this is 'not right, but also not wrong'. Not bad for v1.

**Comment by jayvdb** [comment]
ack d936d16

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**